packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts

```diff
@@ -183,12 +205,27 @@ it('Create a directly funded channel between two wallets ', async () => {
     turnNum: 4,
   });
 
-  expect(await closeCompletedA).toMatchObject({channelId, objectiveType: 'CloseChannel'});
-  expect(await closeCompletedB).toMatchObject({channelId, objectiveType: 'CloseChannel'});
+  expect(await closeCompletedA).toMatchObject({
```

Comment:
I _think_ this logic is only testing the happy path. There doesn't appear to be coverage in this test file that says:
- This promise does _not_ resolve to a `'CloseChannel'` event after 3 block confirmations

I think you could do something like the following
```
async function rejectIfResolves(promiseThatShouldNotResolve: Promise<any>, duration = 1_000) {
  return new Promise((resolve, reject) => {
    promiseThatShouldNotResolve.then(reject);
    setTimeout(resolve, duration);
  });
}

ganache.mineBlocks(2);
await rejectIfResolves(closeCompletedA);
ganache.mineBlocks(3);
await expect(closeCompletedA).resolves.toMatchObject(snip);
```
to assert that, though this is a little hacky.

Reply:
You are right that the non-happy path is not tested. This test would be quite a bit easier to write once https://github.com/statechannels/statechannels/issues/2928 is resolved. In that case, we can simply:
1. Read channel status. Channel status should be `closed`.
1. Mine 3 blocks. Channel status should still be `closed`.
1. Mine 2 more blocks. Channel status should be `closedAndDefunded`.

Thoughts?

Reply:
Added some TODOs to #2928